audit-connector/src/main/java/com/sequenceiq/cloudbreak/audit/converter/FlowEventDataExtractor.java

```diff
@@ -0,0 +1,94 @@
+    private String cbVersion;
+
+    @Inject
+    private Map<String, FlowResourceAuditEventConverter> resourceAuditEventConverters;
+
+    @Override
+    public EventData eventData(StructuredFlowEvent structuredEvent) {
+        Map<String, Object> eventDetails = new HashMap<>();
+        eventDetails.put("userCrn", structuredEvent.getOperation().getUserCrn());
+        eventDetails.put("clusterCrn", structuredEvent.getOperation().getResourceCrn());
+        eventDetails.put("timestamp", System.currentTimeMillis());
+        eventDetails.put("environmentCrn", structuredEvent.getOperation().getEnvironmentCrn());
+        eventDetails.put("flowState", getFlowStat(structuredEvent));
+        eventDetails.put("flowId", structuredEvent.getFlow().getFlowId());
+        return ServiceEventData.builder()
+                .withEventDetails(new Json(eventDetails).getValue())
+                .withVersion(cbVersion)
+                .build();
+    }
+
+    private String getFlowStat(StructuredFlowEvent structuredEvent) {
+        String flowState = structuredEvent.getFlow().getFlowState();
+        return "INIT_STATE".equals(flowState) ? flowState : structuredEvent.getFlow().getNextFlowState();
+    }
+
+    @Override
+    public AuditEventName eventName(StructuredFlowEvent structuredEvent) {
+        FlowDetails flow = structuredEvent.getFlow();
+        String flowEvent = flow.getFlowEvent();
+        AuditEventName eventName = null;
+        String resourceType = structuredEvent.getOperation().getResourceType();
+        FlowResourceAuditEventConverter flowResourceAuditEventConverter = getConverter(resourceType);
+        if (flowResourceAuditEventConverter != null) {
+            eventName = flowResourceAuditEventConverter.auditEventName(structuredEvent);
+        }
+        if (eventName != null) {
+            return eventName;
+        }
+        String flowState = flow.getFlowState();
+        String flowType = flow.getFlowType();
+        throw new UnsupportedOperationException(String.format("The %s, %s and %s does not support for auditing for %s",
+                flowType, flowEvent, flowState, resourceType));
+    }
+
+    @Override
+    public Crn.Service eventSource(StructuredFlowEvent structuredEvent) {
+        return Crn.fromString(structuredEvent.getOperation().getResourceCrn()).getService();
+    }
+
+    @Override
+    public String sourceIp(StructuredFlowEvent structuredEvent) {
+        return null;
+    }
+
+    @Override
+    public boolean shouldAudit(StructuredEvent structuredEvent) {
+        OperationDetails operation = structuredEvent.getOperation();
+        FlowResourceAuditEventConverter flowResourceAuditEventConverter = getConverter(operation.getResourceType());
+        if (flowResourceAuditEventConverter == null) {
+            return false;
+        }
+        boolean crn = Crn.isCrn(operation.getResourceCrn());
+        return flowResourceAuditEventConverter.shouldAudit((StructuredFlowEvent) structuredEvent) && crn;
```

Comment:
swap the sides of  `&&` as it is a short-circuit operator and `crn` value is already known